apps/omg_watcher/lib/omg_watcher/web/controllers/transaction.ex

```diff
@@ -47,10 +47,27 @@ defmodule OMG.Watcher.Web.Controller.Transaction do
   Submits transaction to child chain
   """
   def submit(conn, params) do
-    with {:ok, tx} <- expect(params, "transaction", :hex) do
-      API.Transaction.submit(tx)
-      |> api_response(conn, :submission)
+    elem =
+      with {:ok, tx} <- expect(params, "transaction", :hex) do
```

Comment:
why use a with in this case? you're calling one function and not a series of steps
```
result = case expect(params, "transaction", :hex) do
{:ok, tx}->api_response(API.Transaction.submit(tx)conn, :submission)
error -> error
end
case result do
whatever....
end
```

Reply:
I select here `with` because it is shorter

Reply:
just copying the elixir handbook... `with clause is used to combine matching clauses`